core/src/test/java/org/elasticsearch/common/logging/DeprecationLoggerTests.java

```diff
@@ -158,4 +164,25 @@ public void testFailsWhenRemovingUnknownThreadContext() throws IOException {
         }
     }
 
+    public void testWarningValueFromWarningHeader() throws InterruptedException {
+        final String s = randomAsciiOfLength(16);
+        final String first = DeprecationLogger.formatWarning(s);
+        // force the clock forward by a second so the dates on the warning values are different
+        Thread.sleep(1000);
```

Comment:
:( . Maybe add a variant of formatWarning where you can supply the time? alternatively we can make sure what we extract is the value of `s` which is probably a better test anyway (we want the original, not that we consistently break things). It seems we don't use escaped chars anyway